Description: We use cmake for autopkgtest
 for as-installed package testing.
 This patch is only for Debian autopkgtest.
 Not for upstream.

Index: qrencode/tests/CMakeLists.txt
===================================================================
--- qrencode.orig/tests/CMakeLists.txt
+++ qrencode/tests/CMakeLists.txt
@@ -1,3 +1,14 @@
+cmake_minimum_required(VERSION 3.21.2)
+project(qrencodetest)
+
+set(CMAKE_THREAD_PREFER_PTHREAD ON)
+find_package(Threads)
+find_package(PNG)
+find_package(Iconv)
+check_include_file(sys/time.h HAVE_SYS_TIME_H)
+check_include_file(time.h     HAVE_TIME_H    )
+check_include_file(pthread.h  HAVE_PTHREAD_H )
+
 add_library(common
     common.c common.h)
 target_link_libraries(common qrencode)
@@ -48,12 +59,12 @@ MAKE_TEST(test_bitstream)
 MAKE_TEST(test_estimatebit)
 MAKE_TEST(test_split)
 
-if(TARGET ICONV::ICONV)
+if(TARGET Iconv::Iconv)
     add_library(decoder STATIC
         decoder.c decoder.h 
         datachunk.c datachunk.h 
         rsecc_decoder.c rsecc_decoder.h)
-    target_link_libraries(decoder ICONV::ICONV)
+    target_link_libraries(decoder Iconv::Iconv)
 
     MAKE_TEST(test_qrinput decoder)
     MAKE_TEST(test_qrspec decoder)
